### PR TITLE
feat: Add support for CycloneDx v1.4

### PR DESCRIPTION
**Pull request by carlosthe19916**

### CODE REVIEW

File: bombastic/index/Cargo.toml [hunk 1]
```diff
@@ -5,7 +5,7 @@ edition = "2021"
 
 [dependencies]
 bombastic-model = { path = "../model" }
-cyclonedx-bom = "0.4.0"
+cyclonedx-bom = { git = "https://github.com/CycloneDX/cyclonedx-rust-cargo", rev = "0aa9ee0c9b28d6b92772330baea957d67f500670" }
```
Comment (ctron) [inline bombastic/index/Cargo.toml:8]: It makes more sense to use the `patch` section of the workspace level Cargo.toml  file.

File: bombastic/model/src/data.rs [hunk 2]
```diff
@@ -2,15 +2,15 @@ use std::fmt::Formatter;
...
+    #[cfg(feature = "cyclonedx")]
     CycloneDX(cyclonedx_bom::prelude::Bom),
     #[cfg(feature = "spdx-rs")]
     SPDX(spdx_rs::models::SPDX),
 }
 
 #[derive(Debug, Default)]
 pub struct Error {
-    #[cfg(feature = "cyclonedx-bom")]
+    #[cfg(feature = "cyclonedx")]
```
Comment (ctron) [inline bombastic/model/src/data.rs:13]: `cyclonedx-bom` is actually the name/ID of the crate. I think we should/have to keep that.

Comment (carlosthe19916) [reply bombastic/model/src/data.rs:13]: Thanks for making a comment in this line. I admit I'm not very familiar with the `#[cfg` thing. I did tried to keep `cyclonedx-bom` but then this following line was never executed but always skipped; even the IDE alerted me that block of code will be disabled as the feature `cyclonedx` was disabled.

https://github.com/trustification/trustification/blob/b7c1230b113f6cf856c4b5a48f0071baff2715ae/bombastic/model/src/data.rs#L67

Are you sure this line must be `cyclonedx-bom` and not `cyclonedx` as it is currently in my PR? 
> This #[cfg] thing is a "must to learn" this week for me :)

Comment (carlosthe19916) [reply bombastic/model/src/data.rs]: Done, thanks!

File: spog/ui/crates/components/src/cyclonedx/mod.rs [hunk 3]
```diff
@@ -15,7 +15,7 @@ pub fn cyclonedx_meta(bom: &cyclonedx_bom::prelude::Bom) -> Html {
         .metadata
         .as_ref()
         .and_then(|m| m.component.as_ref())
-        .map(|c| c.version.to_string());
+        .map(|c| c.version.as_ref().map(|e| e.to_string()).unwrap_or("".to_string()));
```
Comment (ctron) [inline spog/ui/crates/components/src/cyclonedx/mod.rs:18]: Works! But I think it would also be possible to use `.unwrap_or_default()` in this case.

**Review by ctron — APPROVED** [review]

**Comment by carlosthe19916** [comment]
@ctron I changed the status of the PR to "ready to be reviewed". The idea is to not wait until https://github.com/CycloneDX/cyclonedx-rust-cargo releases a new version but just to point to the latest commit within their main branch.

-  I created a PR in their repository for fixing the bug I've found and reported yesterday https://github.com/CycloneDX/cyclonedx-rust-cargo/pull/616

What do you think? Should we wait until https://github.com/CycloneDX/cyclonedx-rust-cargo/pull/616 is reviewed and merged? or we can proceed with this PR as it is so we have support for Cyclonedx v.1.4?

File: spog/ui/crates/components/src/cyclonedx/mod.rs (hunk 3 above)
Comment (carlosthe19916) [reply spog/ui/crates/components/src/cyclonedx/mod.rs]: Done! Thanks for pointing that out

File: spog/ui/src/pages/scanner/mod.rs [hunk 4]
```diff
@@ -189,7 +189,7 @@ fn common_header(props: &CommonHeaderProperties) -> Html {
                     <Content>
                         <Title>{"Scan an SBOM"}</Title>
                         <p>
-                            {"Load an existing CycloneDX 1.3 or SPDX 2.2 file"}
+                            {"Load an existing CycloneDX 1.3,1.4 or SPDX 2.2 file"}
```
[the remark this answers is not in the document]
Comment (i386x) [reply spog/ui/src/pages/scanner/mod.rs:192]: I noticed that I forgot to submit my review so doing it additionally now to keep the record in GitHub.

**Pull request merged**